### PR TITLE
feat: Support emqx stats to statsd server

### PR DESCRIPTION
**Pull request by DDDHuang**
Fixes <issue-number>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

## More information

### CODE REVIEW

**Review by turtleDeng — APPROVED** [review]

File: apps/emqx_statsd/src/emqx_statsd_sup.erl [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+ init([]) ->
+    {ok, {{one_for_one, 10, 100}, []}}.
+
+start_statsd() ->
+    {ok, Pid} = supervisor:start_child(?MODULE, estatsd_child_spec()),
+    {ok, _Pid1} = supervisor:start_child(?MODULE, emqx_statsd_child_spec(Pid)).
+
+stop_statsd() ->
+    ok = supervisor:terminate_child(emqx_statsd_sup, emqx_statsd),
```
Comment (turtleDeng) [inline apps/emqx_statsd/src/emqx_statsd_sup.erl:29]: ?MODULE

File: apps/emqx_statsd/src/emqx_statsd_sup.erl [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    ok = supervisor:terminate_child(emqx_statsd_sup, estatsd).
```
Comment (turtleDeng) [inline apps/emqx_statsd/src/emqx_statsd_sup.erl:30]: ?MODULE

File: apps/emqx_statsd/src/emqx_statsd_sup.erl [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        [Domain, Port] -> {Domain, list_to_integer(Port)}
+    end.
+
+tags(Map) ->
+    Tags = maps:to_list(Map),
+    [{atom_to_binary(Key, utf8), Value} || {Key, Value} <- Tags].
+
+emqx_statsd_child_spec(Pid) ->
+    #{id       => emqx_statsd
+    , start    => {emqx_statsd, start_link, [emqx_statsd_options(Pid)]}
```
Comment (turtleDeng) [inline apps/emqx_statsd/src/emqx_statsd_sup.erl:62]: {emqx_statsd, start_link, [[{estatsd_pid, Pid} | emqx_statsd_options()]]}

File: apps/emqx_statsd/src/emqx_statsd_sup.erl [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    , restart  => permanent
+    , shutdown => 5000
+    , type     => worker
+    , modules  => [emqx_statsd]}.
+
+emqx_statsd_options(Pid) ->
```
Comment (turtleDeng) [inline apps/emqx_statsd/src/emqx_statsd_sup.erl:68]: emqx_statsd_options() ->

File: apps/emqx_statsd/src/emqx_statsd_sup.erl [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    SampleTimeInterval = get_conf(sample_time_interval, ?DEFAULT_SAMPLE_TIME_INTERVAL),
+    FlushTimeInterval = get_conf(flush_time_interval, ?DEFAULT_FLUSH_TIME_INTERVAL),
+    [{estatsd_pid, Pid}, {sample_time_interval, SampleTimeInterval}, {flush_time_interval, FlushTimeInterval}].
```
Comment (turtleDeng) [inline apps/emqx_statsd/src/emqx_statsd_sup.erl:71]: [{sample_time_interval, SampleTimeInterval}, {flush_time_interval, FlushTimeInterval}].

**Review by turtleDeng — APPROVED** [review]

**Review by turtleDeng — APPROVED** [review]

**Comment by zmstone** [comment]
Please add tests.
And a question, how is the config loaded?

File: apps/emqx_statsd/src/emqx_statsd_sup.erl [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+estatsd_options() ->
+    Server = get_conf(server, {?DEFAULT_HOST, ?DEFAULT_PORT}),
+    {Host, Port} = host_port(Server),
+    Prefix = get_conf(prefix, ?DEFAULT_PREFIX),
+    Tags = tags(get_conf(tags, ?DEFAULT_TAGS)),
+    BatchSize = get_conf(batch_size, ?DEFAULT_BATCH_SIZE),
+    [{host, Host}, {port, Port}, {prefix, Prefix}, {tags, Tags}, {batch_size, BatchSize}].
+
+host_port({Host, Port}) -> {Host, Port};
```
Comment (zmstone) [inline apps/emqx_statsd/src/emqx_statsd_sup.erl:49]: Make it a `converter` of the `server` field

File: apps/emqx_statsd/src/emqx_statsd_schema.erl [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+server(not_nullable) -> true;
+server(_) -> undefined.
+
+prefix(type) -> string();
+prefix(default) -> "emqx";
+prefix(not_nullable) -> false;
+prefix(_) -> undefined.
+
+batch_size(type) -> integer();
+batch_size(not_nullable) -> true;
```
Comment (zmstone) [inline apps/emqx_statsd/src/emqx_statsd_schema.erl:31]: Should be `nullable` (false) , not `not_nullable`

**Review by zmstone — COMMENTED** [review]
Sorry for the late review.

Please follow up the changes with a new pr

**Pull request merged**